src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  const sourceArray = sourceString.split(';');
+
+  const objectProperties = sourceArray.reduce((acc, property) => {
+    const propertyArray = property.trim().split(':', 2);
+    const [keyName, keyValue] = propertyArray;
+
+    if (keyName && keyValue !== undefined) {
```

Comment:
The condition ``keyName && keyValue !== undefined`` is correct, but `keyValue` cannot be undefined because 
``split(':', 2)`` always returns an array with two elements. Better to check if `keyValue` is a non-empty string.

```suggestion

// Check if the key is not an empty string and if the value exists and is not an empty string.
    if (keyName && keyValue) {
```